lohica-js-network/src/components/inputComponents/FormComponent.js

```diff
@@ -0,0 +1,180 @@
+            data.append('email', email);
+            data.append('gender', gender);
+            data.append('age', age);
+
+            // beginning of fetch
+
+            fetch(signupRoute,{
+                method: 'POST',
+                body: data
+            }).then((mes) => mes.json())
+                .then(data => {
+                    // show success or warning alert depending on user got registered or not
+                    switch(data.statusCode){
+                        case 200:
+                            this.props.dispatch(regValidHandle({
+                                password: data.userPass,
+                                message: "You have successfully registered, here's your password: ",
+                                show: true,
+                                alertStyle: "success"
+                            }));
+                            this.props.dispatch(successReg({
+                                regStatus: true
+                            }));                   
+                            break;
+                        case 409:
+                            this.props.dispatch(regValidHandle({
+                                password: "",
+                                message: "User with that email already exists",
+                                show: true,
+                                alertStyle: "danger"
+                            }));
+                            break;
+                        case 400: 
+                            this.props.dispatch(regValidHandle({
+                                password: "",
+                                message: data.message,
+                                show: true,
+                                alertStyle: "warning"
+                            }));
+                            break;
+                        default:
+                            break
+                            
+                    }  
+                    
+                    this.props.dispatch(regValidHideMes({showWarning: false}));                  
+                })
+                .catch(err => console.log(err));
+
+
+        } else {
+            this.props.dispatch(regValidShowMes({showWarning: true, show: false}));
+        }
+    }
+
+
+    render() { 
+        let regValidStore = this.props.inputState.regForm.regValidateState;
+
+        let alert = <Alert bsStyle={regValidStore.alertStyle}>
+                    <h3>
+                        {regValidStore.message}<strong>{this.props.inputState.regForm.regValidateState.password}</strong>
+                    </h3>
+                </Alert>;
+
+        let alertAllFields = <Alert bsStyle="warning">
+                    <h4>
+                        Please fill in all the fileds marked with '*'
+                    </h4>
+                </Alert>;
+
+        // age options
+        let Options = [];
+        for(let i =1; i<=99; i++){
+            Options.push(`${i}`);
```

Comment:
I've seen this code before? is it needed?

Reply:
that piece of code is taken out to the utils folder and imported